azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -48,11 +48,14 @@ def __init__(
         problem_type: ProblemType = ProblemType.ising
     ):
         self.name = name
-        self.terms = terms if terms is not None else []
+        self.terms = terms.copy() if terms is not None else []
```

Comment:
It's copied in the constructor because the add_term/add_terms methods append terms to the list - modifying the same refence of the list that you passed in which can cause unexpected behavior for the user who may expect the array that they passed in to not have been modified (I encountered this).